clis/teliod/src/daemon.rs

```diff
@@ -25,19 +31,31 @@ pub enum TelioTaskCmd {
     UpdateMeshmap(MeshMap),
     // Get telio status
     GetStatus(oneshot::Sender<TelioStatusReport>),
+    // Enable or disable meshnet
+    SetMeshnet(bool, oneshot::Sender<Result<(), TeliodError>>),
+    // Configure the system interface
+    SetIp(oneshot::Sender<Result<(), TeliodError>>),
     // Break the receive loop to quit the daemon and exit gracefully
     Quit,
 }
 
 const EMPTY_TOKEN: &str = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
 
+#[derive(Debug, Default)]
+pub struct TelioTaskStates {
+    interface_ip_address: Option<IpAddr>,
+    meshnet_is_enabled: bool,
+}
+
 // From async context Telio needs to be run in separate task
 fn telio_task(
     node_identity: Arc<DeviceIdentity>,
     auth_token: Arc<String>,
     mut rx_channel: mpsc::Receiver<TelioTaskCmd>,
     tx_channel: mpsc::Sender<TelioTaskCmd>,
     interface_name: &str,
+    auto_start_meshnet: bool,
+    auto_configure_interface: bool,
 ) -> Result<(), TeliodError> {
     debug!("Initializing telio device");
     let mut telio = Device::new(
```

Comment:
shouldn't this `telio` instance be created only when Enable meshnet?

Reply:
Please refer to this [comment](https://github.com/NordSecurity/libtelio/pull/953#discussion_r1853572042)